libs/pika/async_mpi/src/check_openmpi_continuations.cpp

```diff
@@ -0,0 +1,20 @@
+//  Copyright (c) 2021 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// simple compile test to see if mpix continuuations are present
+// returns 0 when available, 0 otherwise
+
+#include <mpi.h>
+
+#if __has_include(<mpi-ext.h>)
+# include <mpi-ext.h>
+#endif
+
+#ifdef OMPI_HAVE_MPI_EXT_CONTINUE
+int main() { return 0; }
+#else
+int main() { return 1; }
+#endif
```

Comment:
```suggestion
//  Copyright (c) 2024 ETH Zurich
//
//  SPDX-License-Identifier: BSL-1.0
//  Distributed under the Boost Software License, Version 1.0. (See accompanying
//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)

#include <mpi.h>
#include <mpi-ext.h>

#if !defined(OMPI_HAVE_MPI_EXT_CONTINUE)
static_assert(false);
#endif

// I forget if we need a main or not here, remove if not necessary.
int main() {}
```
If we don't have MPI this will fail already at the includes. If we don't have OpenMPI this will also fail at the `mpi-ext.h` include. The `static_assert` takes care of the rest. This way we don't have to run it and check the result. What do you think?

Reply:
I'm updated the checks to use that. Tested it locally and it is fine, thanks